iwyu_test_util.py

```diff
@@ -73,12 +74,56 @@ class Features:
   # Parses the output of 'include-what-you-use -print-targets'.
   SUPPORTED_TARGET_RE = re.compile(r'^\s+([a-z0-9_-]+)\s* - .*$')
 
+  # Parses the output of 'include-what-you-use -dM -E'.
+  PREPROCESSOR_MACRO_RE = re.compile(r'^#define ([A-z_][A-z0-9_]*)')
+
   def __init__(self):
     # Maps all known features names to prerequisite predicates.
     self.known = {
-      'has-target': Features.HasTarget
+      'has-target': Features.HasTarget,
+      'using-cstdlib': Features.UsingCstdlib,
+      'using-cxxstdlib': Features.UsingCXXstdlib,
     }
 
+  @staticmethod
+  def _GetPreprocessorMacros(lang, header):
+    iwyu = _GetIwyuPath()
+    txt = f'#include {header}'
+
+    cmd = [iwyu, '-dM', '-E', '-x', lang]
+    # Take into account extra arguments specified in IWYU_EXTRA_ARGS -
+    # the user may be trying to test with different includes, standard
+    # libraries or whatever.
+    #
+    # For feature tests we _don't_ take into account test-specific
+    # IWYU_ARGS.
+    #  * the argument could be the thing that breaks things
+    #  * the feature result should not be test specific
+    env_iwyu_extra_args = os.getenv('IWYU_EXTRA_ARGS')
+    if env_iwyu_extra_args:
+      cmd += shlex.split(env_iwyu_extra_args)
```

Comment:
Seems worth it to extract this into a function that we can reuse between here and `TestIwyuOnRelativeFile`